Examples/Io/Csv/src/CsvMultiTrajectoryWriter.cpp

```diff
@@ -0,0 +1,207 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Io/Csv/CsvMultiTrajectoryWriter.hpp"
+
+#include "Acts/EventData/MultiTrajectoryHelpers.hpp"
+#include "ActsExamples/Utilities/Paths.hpp"
+#include "ActsExamples/Validation/TrackClassification.hpp"
+
+#include <ios>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+using namespace ActsExamples;
+
+CsvMultiTrajectoryWriter::CsvMultiTrajectoryWriter(
+    const CsvMultiTrajectoryWriter::Config& cfg, Acts::Logging::Level level)
+    : WriterT<TrajectoriesContainer>(cfg.inputTrajectories,
+                                     "CsvMultiTrajectoryWriter", level),
+      m_cfg(cfg) {
+  if (m_cfg.inputTrajectories.empty()) {
+    throw std::invalid_argument("Missing input trajectories collection");
+  }
+}
+
+ProcessCode CsvMultiTrajectoryWriter::writeT(
+    const AlgorithmContext& context,
+    const TrajectoriesContainer& trajectories) {
+  // open per-event file
+  std::string path =
+      perEventFilepath(m_cfg.outputDir, "CKFtracks.csv", context.eventNumber);
+  std::ofstream mos(path, std::ofstream::out | std::ofstream::trunc);
+  if (!mos) {
+    throw std::ios_base::failure("Could not open '" + path + "' to write");
+  }
+
+  using HitParticlesMap = ActsExamples::IndexMultimap<ActsFatras::Barcode>;
+  const auto& hitParticlesMap = context.eventStore.get<HitParticlesMap>(
+      m_cfg.inputMeasurementParticlesMap);
+
+  struct trackInfo {
```

Comment:
I guess it might be better to define this struct in the header file? Also, this can be inherited from the `TrajectoryState` to reduce some redundancy, right?

Reply:
Done